hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -1268,13 +1268,29 @@ public void decrementOpDepth() {
         props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class.getName());
         props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, ExternalizableDeserializer.class.getName());
 
-        KafkaConsumer<Integer, Externalizable> consumer = new KafkaConsumer<Integer, Externalizable>(props);
-        List ps = consumer.partitionsFor(topicName);
-        List<Integer> partitions = new ArrayList<>(ps.size());
-        for (int i = 0; i < ps.size(); ++i) {
-            partitions.add(i);
+        List<Integer> partitions;
+        String topicName;
```

Comment:
Is this our naming convention for kafka partition?

Reply:
Often the first batch received by SSDS at the beginning of a data run does not have data in every partition.  In this case, to the topic name SSDS appends :: followed by a comma-separated list of the partition numbers that have data, so that the consumer in the DB will not poll empty partitions, which results in long timeouts.

Kafka's convention for a partition name is topic name hyphen partition number.